src/Microsoft.Net.Http.Headers/DateTimeFormatter.cs

```diff
@@ -0,0 +1,161 @@
+        private static readonly byte[] SatBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Saturday));
+        private static readonly byte[] SunBytes = UTF8.GetBytes(GetDayName(DayOfWeek.Sunday));
+
+        private static readonly byte[] JanBytes = UTF8.GetBytes(GetMonthName(1));
+        private static readonly byte[] FebBytes = UTF8.GetBytes(GetMonthName(2));
+        private static readonly byte[] MarBytes = UTF8.GetBytes(GetMonthName(3));
+        private static readonly byte[] AprBytes = UTF8.GetBytes(GetMonthName(4));
+        private static readonly byte[] MayBytes = UTF8.GetBytes(GetMonthName(5));
+        private static readonly byte[] JunBytes = UTF8.GetBytes(GetMonthName(6));
+        private static readonly byte[] JulBytes = UTF8.GetBytes(GetMonthName(7));
+        private static readonly byte[] AugBytes = UTF8.GetBytes(GetMonthName(8));
+        private static readonly byte[] SepBytes = UTF8.GetBytes(GetMonthName(9));
+        private static readonly byte[] OctBytes = UTF8.GetBytes(GetMonthName(10));
+        private static readonly byte[] NovBytes = UTF8.GetBytes(GetMonthName(11));
+        private static readonly byte[] DecBytes = UTF8.GetBytes(GetMonthName(12));
+
+        private static readonly byte[] GmtBytes = UTF8.GetBytes("GMT");
+
+        // The format is "ddd, dd MMM yyyy HH:mm:ss GMT".
+        private const int Rfc1123DateLength = 29;
+
+        // ASCII numbers are in the range 48 - 57.
+        private const int AsciiNumberOffset = 0x30;
+
+        private const char Comma = ',';
+        private const char Space = ' ';
+        private const char Colon = ':';
+
+        public static unsafe string ToRfc1123String(this DateTimeOffset dateTime)
+        {
+            var offset = 0;
+            char* target = stackalloc char[Rfc1123DateLength];
+            var universalDateTime = dateTime.ToUniversalTime();
+
+            offset = FormatDayOfWeek(universalDateTime.DayOfWeek, target, offset);
+            offset = Format(Comma, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatNumber(universalDateTime.Day, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatMonth(universalDateTime.Month, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatYear(universalDateTime.Year, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatTimeOfDay(universalDateTime.TimeOfDay, target, offset);
+            offset = Format(Space, target, offset);
+            offset = Format(GmtBytes, target, offset);
+
+            return new string(target, 0, offset);
+        }
+
+        private static unsafe int FormatDayOfWeek(DayOfWeek dayOfWeek, char* target, int offset)
+        {
+            switch (dayOfWeek)
+            {
+                case DayOfWeek.Sunday: return Format(SunBytes, target, offset);
```

Comment:
Yeah, sure. I guess we could just reuse [`DateTimeFormatInfo.AbbreviatedDayNames`](https://msdn.microsoft.com/en-us/library/system.globalization.datetimeformatinfo.abbreviateddaynames(v=vs.110).aspx) and [`DateTimeFormatInfo.AbbreviatedMonthNames`](https://msdn.microsoft.com/en-us/library/system.globalization.datetimeformatinfo.abbreviatedmonthnames(v=vs.110).aspx)